scripts/updateConfig.js

```diff
@@ -7,70 +7,63 @@ const fs = require('fs');
-    modifyPackageJsonForNonProduction(packageJson, nightlyBuildName, nightlyBuildDisplayName, nightlyBuildPublisher);
-  } else if (process.env.TRAVIS_TAG) {
-    const isProduction = /^v?[0-9]+\.[0-9]+\.[0-9]+$/.test(process.env.TRAVIS_TAG || '');
-    const isTestVersion = /^v?[0-9]+\.[0-9]+\.[0-9]+-[rR][cC]/.test(process.env.TRAVIS_TAG || '');
+// Nightly Build
+if (process.env.BUILD_REASON === "Schedule") {
+  const nightlyBuildName = "test-owl-project-nightly";
+  const nightlyBuildDisplayName = "Test OWL Project (Nightly)";
+  updateConfigForNonProduction(packageJson, nightlyBuildName, nightlyBuildDisplayName);
+} else if (process.env.IS_PROD) {
+  // Update resource link
+  const codeGenUrl = "https://aka.ms/iot-codegen-cli-for-workbench";
+  packageJson.codeGenConfigUrl = codeGenUrl;
 
-    if (isProduction) {
-      // Update resource link
-      const codeGenUrl = "https://aka.ms/iot-codegen-cli-for-workbench";
-      packageJson.codeGenConfigUrl = codeGenUrl;
-
-      // Update production AI Key
-      packageJson.aiKey = process.env['PROD_AIKEY'];
-    } else if (isTestVersion) {
-      const testName = "test-owl-project";
-      const testDisplayName = "Test OWL Project RC";
-      const testPublisher = "IoTDevExBuild";
-      modifyPackageJsonForNonProduction(packageJson, testName, testDisplayName, testPublisher);
-
-      // Modify extensionId in template files
-      const extensionIdPattern = /vsciot-vscode.vscode-iot-workbench/g;
-      const rcExtensionId = 'iotdevexbuild.test-owl-project';
-
-      const arm7DevcontainerJsonFile = "resources/templates/arm7/devcontainer.json";
-      const arm8DevcontainerJsonFile = "resources/templates/arm8/devcontainer.json";
-      const x86DevcontainerJsonFile = "resources/templates/x86/devcontainer.json";
-      const files = [arm7DevcontainerJsonFile, arm8DevcontainerJsonFile, x86DevcontainerJsonFile];
-      files.forEach(filePath => {
-        const originalJsonFile = fs.readFileSync(filePath).toString();
-        const replaceJson = originalJsonFile.replace(extensionIdPattern, rcExtensionId);
-        fs.writeFileSync(filePath, replaceJson);
-      });
-    }
-  }
-
-  fs.writeFileSync('package.json', JSON.stringify(packageJson, null, 2) + '\n');
+  // Update production AI Key
+  packageJson.aiKey = process.env.PROD_AIKEY;
+} else if (process.env.IS_TEST) {
+  const testName = "test-owl-project";
+  const testDisplayName = "Test OWL Project RC";
+  updateConfigForNonProduction(packageJson, testName, testDisplayName);
 }
 
+fs.writeFileSync('package.json', JSON.stringify(packageJson, null, 2) + '\n');
+
 /**
  * Update package.json with test name, displayName, publisher, ai aky.
  * Trim version number. Delete extension icon.
+ * Update extension id in template files.
  * @param {*} packageJson package json oject
  * @param {*} testName test extension name
  * @param {*} testDisplayName test extension displate name
  * @param {*} testPublisher test publisher
```

Comment:
Please remove this parameter definition of testPublisher

Reply:
Fixed.